joblib/test/test_numpy_pickle.py

```diff
@@ -217,40 +181,39 @@ def test_numpy_persistence():
 
         # Test with an object containing multiple numpy arrays
         obj = ComplexTestObject()
-        filenames = numpy_pickle.dump(obj, this_filename,
-                                      compress=compress)
+        filenames = numpy_pickle.dump(obj, filename, compress=compress)
         # All is cached in one file
         assert len(filenames) == 1
 
-        obj_loaded = numpy_pickle.load(this_filename)
+        obj_loaded = numpy_pickle.load(filename)
         assert isinstance(obj_loaded, type(obj))
         np.testing.assert_array_equal(obj_loaded.array_float, obj.array_float)
         np.testing.assert_array_equal(obj_loaded.array_int, obj.array_int)
         np.testing.assert_array_equal(obj_loaded.array_obj, obj.array_obj)
 
 
 @with_numpy
-def test_numpy_persistence_bufferred_array_compression():
+def test_numpy_persistence_bufferred_array_compression(tmpdir):
     big_array = np.ones((_IO_BUFFER_SIZE + 100), dtype=np.uint8)
-    filename = env['filename'] + str(random.randint(0, 1000))
+    filename = tmpdir.join('test.pkl').strpath
     numpy_pickle.dump(big_array, filename, compress=True)
     arr_reloaded = numpy_pickle.load(filename)
 
     np.testing.assert_array_equal(big_array, arr_reloaded)
 
 
 @with_numpy
-def test_memmap_persistence():
+def test_memmap_persistence(tmpdir):
     rnd = np.random.RandomState(0)
     a = rnd.random_sample(10)
-    filename = env['filename'] + str(random.randint(0, 1000))
+    filename = tmpdir.join('test1.pkl').strpath
     numpy_pickle.dump(a, filename)
     b = numpy_pickle.load(filename, mmap_mode='r')
 
     assert isinstance(b, np.memmap)
 
     # Test with an object containing multiple numpy arrays
-    filename = env['filename'] + str(random.randint(0, 1000))
+    filename = tmpdir.join('test2.pkl').strpath
```

Comment:
Restored these lines back and it's solved. Sadly I can't reproduce errors on Windows locally.